src/integrationTest/java/uk/gov/hmcts/darts/common/controller/CourthouseApiTest.java

```diff
@@ -0,0 +1,228 @@
+            .andExpect(jsonPath("$.code", is(761)))
+            .andExpect(jsonPath("$.created_date_time", is(notNullValue())))
+            .andExpect(jsonPath("$.last_modified_date_time", is(notNullValue())));
+    }
+
+    /**
+     * Test adds courthouse and then retrieves to check for equality.
+     */
+    @Test
+    void courthousesGetNonExistingId() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = get("/courthouses/{courthouse_id}", 900)
+            .contentType(MediaType.APPLICATION_JSON_VALUE);
+        mockMvc.perform(requestBuilder).andExpect(status().isNotFound());
+    }
+
+    /**
+     * Test adds multiple courthouses and then retrieves them all to check for equality.
+     */
+    @Test
+    void courthousesGetAll() throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+
+        MvcResult haverfordwestResponse = makeRequestToAddCourthouseToDatabase(
+            REQUEST_BODY_HAVERFORDWEST_JSON);
+        MvcResult swanseaResponse = makeRequestToAddCourthouseToDatabase(REQUEST_BODY_SWANSEA_JSON);
+
+
+        MockHttpServletRequestBuilder requestBuilder = get("/courthouses")
+            .contentType(MediaType.APPLICATION_JSON_VALUE);
+
+        MvcResult response = mockMvc.perform(requestBuilder).andExpect(status().isOk()).andDo(print()).andReturn();
+
+        List<ExtendedCourthouse> courthouseList = objectMapper.readValue(response.getResponse().getContentAsString(), new TypeReference<>() {});
+        ExtendedCourthouse haverfordwestCourthouse = objectMapper.readValue(haverfordwestResponse.getResponse().getContentAsString(), ExtendedCourthouse.class);
+        ExtendedCourthouse swanseaCourthouse = objectMapper.readValue(swanseaResponse.getResponse().getContentAsString(), ExtendedCourthouse.class);
+
+        assertTrue(courthouseList.contains(swanseaCourthouse));
+        assertTrue(courthouseList.contains(haverfordwestCourthouse));
+    }
+
+    /**
+     * Test adds courthouse and checks if returned object matches.
+     */
+    @Test
+    void courthousesPost() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = post("/courthouses")
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .content(getContentsFromFile(REQUEST_BODY_HAVERFORDWEST_JSON));
+        mockMvc.perform(requestBuilder).andExpect(status().isCreated())
+            .andExpect(jsonPath("$.id", is(notNullValue())))
+            .andExpect(jsonPath("$.courthouse_name", is("HAVERFORDWEST")))
+            .andExpect(jsonPath("$.code", is(761)))
+            .andExpect(jsonPath("$.created_date_time", is(notNullValue())))
+            .andExpect(jsonPath("$.last_modified_date_time", is(notNullValue())));
+    }
+
+    /**
+     * Test adds courthouse and checks if returned object matches.
```

Comment:
No need to add comments to every test. The test name should imply what the test is doing. If not clear, then please update. A good pattern is normally 'givenMyTestHasThisInput_thenTheResultIsThis()'.

Reply:
I am happy to delete the comments I think the test names are self explanatory.

Reply:
ACCEPT, will remove comments. most tests are self-explanatory.